### ISSUE TITLE
question regarding dependency constraint

### ISSUE THREAD

**Issue by jasdeepsingh** [tenex/rails-assets#364]
`bower.json` of [angular-ui/angular-google-maps@2.3.0](https://github.com/angular-ui/angular-google-maps/blob/2.3.0/bower.json) specifies a dependency on `lodash`: `3.X`

```
"dependencies": {
    "angular": "1.2 - 1.4",
    "angular-simple-logger": ">=0.0.1",
    "lodash": "3.X"
  }
```

In my `Gemfile`, I have: 

```
gem 'rails-assets-angular-google-maps', '~> 2.3.0'
gem 'rails-assets-lodash'
```

But I consistently get the following error, every time I do `bundle` or `bundle update`

```
Bundler could not find compatible versions for gem "rails-assets-lodash":
  In Gemfile:
    rails-assets-lodash

rails-assets-angular-google-maps (~> 2.3.0) was resolved to 2.3.0, which depends
on
      rails-assets-lodash (= 3.X)

Could not find gem 'rails-assets-lodash (= 3.X)', which is required by gem
'rails-assets-angular-google-maps (~> 2.3.0)', in any of the sources.
```

What version specifier should I be using for `rails-asset-lodash` to meet the `3.X` requirement?

So far I've tried various variations, but all lead to the same error. 

```
gem 'rails-assets-lodash', '= 3.X'
gem 'rails-assets-lodash', '3.0.0'
gem 'rails-assets-lodash', '= 3.0.0'
gem 'rails-assets-lodash', '= 3.10.1'
gem 'rails-assets-lodash', '3.10.1'
```

Is this something that should be changed in `angular-google-maps`'s `bower.json` file or this can be handled in my own `Gemfile`?

**Comment by hut8** [comment]
Looks like the code checks for a `x` but not a capital `X`. To be fair, that's what the spec dictates, but PRs are welcome.

**Comment by jasdeepsingh** [comment]
@hut8 Opened a PR (#364) for this. Looking forward for your reviewal. Thanks!

**Comment by hut8** [comment]
Thanks so much for the PR. It's now merged and deployed. Is the issue resolved?

**Comment by jasdeepsingh** [comment]
@hut8 Awesome! Thanks so much for merging and deploying this, I'll try to update my Gemfile shortly and verify the fix. Thanks again for Rails-Assets, always handy.

**Comment by jasdeepsingh** [comment]
~~@hut8 the issue still exists for me, I was under the impression that if I issue a request to rebuild the component via the [Add Component](https://rails-assets.org/components/new) page from the site, that should update the `dependencies` column in the `versions` table in the DB, turns out I was wrong. (had to further test this locally via the web UI) 😥~~

~~Running `bin/rake 'component:convert[angular-google-maps,2.3.0]'` from the command line properly updates the dependency graph, Is there any other way I can issue a rebuild for the component or would you accept a PR if `Add Component` took care of this?~~
1. Never mind, I was able to find the `rebuild` route from the routes file 😄 
2. On a side note, I think having a section showing dependencies on a component page + the last updated at timestamp for that version would be helpful for the end users, what do you think of this idea? I can put together a PR in case you feel that would be a welcomed addition.

**Comment by hut8** [comment]
That would be a very welcomed addition! Thanks so much for the thought and time you've put into this. We took out the rebuild button a while ago, but the route as you found is still there :smile: Also, if you just "add" an existing component, it rebuilds the latest version which is a bug I filed separately, but in a pinch it works for rebuilding. What I should have done is asked you to figure out every bower package affected by this and make a "migration", but you've already contributed lots and I didn't want to trouble you.
